NGitLab/Impl/UserClient.cs

```diff
@@ -27,6 +27,21 @@ public Task<User> GetByIdAsync(int id, CancellationToken cancellationToken = def
         return _api.Get().ToAsync<User>(User.Url + "/" + id.ToStringInvariant(), cancellationToken);
     }
 
+    public async Task<User> GetByUserNameAsync(string username, CancellationToken cancellationToken = default)
+    {
+        // This query returns 0 or 1 user.
+        await foreach (var u in _api.Get().GetAllAsync<User>(User.Url + "?username=" + username).WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            return u;
+        }
+
+        throw new GitLabException("User not found.")
```

Comment:
Note: GitLab does _not_ throw when the "search by username" query fails to find a match, but since `GetByIdAsync` throws a 404, it seems reasonable to throw a 404 from this method.